doc/reverse_engineering/game_mechanics/formations-formulas.md

```diff
@@ -0,0 +1,248 @@
+# Units formations
+
+Military units can be arranged in various formations. Here's an overview on how this may be implemented
+
+### Line formation
+
+If you table the number of units in the first row, in the second row and so on accordingly to the number of units in the group, you'll find a pattern in the units number intervals for a given row count: namely, the length of the first interval `(0...6)` is `6`, the second `(7...15)` is `9 = 6+3 + 0`, the third `(16...28)` is `13 = (6+3) + 3+1`, the fourth (maybe) `(29...46)` is `18 = ((6+3) + 3+1 ) + 3+2`.
+
+The value of `46` is chosen by observing that the difference between the last value for a certain row count and the first value for the following row count is always `3`; in this case, with `46` units `4` rows are formed: the first 3 with 12 units and the last with 10. For the next value, `47`, we guess `5` rows with `10, 10, 10, 10, 7`.
+
+Of course, since there's a limit of 40 to the number of selectable units, there's unfortunately no way to check this guess.
+
+##### Formulas
+
+If we neglect the first interval `(0...6)` from above, we can define a recurrence equation between the others.
+It looks like this:
+
+``` python
+def f(n):
+    if n == 0:
+    	return 9
+    return f(n-1) + n + 3
+```
+
+In order to find the nth last value before the row count changes, we just need to sum f(i) from 0 up to n. Then, in order to get the row count for a specified numOfUnits, we can iteratively check which of the values is the upper bound of the interval where our numOfUnits lies:
+
+``` python
+def findRowCount(numOfUnits):
+    # the last numOfUnits before the row count changes
+    lastBeforeInc = 6  
+    # iteration counter
+    i = 0
```

Comment:
If you write Python code, it should adhere to Python style guides, i.e. snake case function/variable names.